src/library/scala/collection/SeqView.scala

```diff
@@ -20,23 +23,49 @@ object SeqView {
   type SomeSeqOps[+A] = SeqOps[A, AnyConstr, _]
 
   /** A view that doesn’t apply any transformation to an underlying sequence */
+  @SerialVersionUID(3L)
   class Id[+A](underlying: SeqOps[A, AnyConstr, _]) extends AbstractSeqView[A] {
     def apply(idx: Int): A = underlying.apply(idx)
     def length: Int = underlying.length
     def iterator: Iterator[A] = underlying.iterator
     override def knownSize: Int = underlying.knownSize
   }
 
+  @SerialVersionUID(3L)
   class Map[+A, +B](underlying: SomeSeqOps[A], f: A => B) extends View.Map[A, B](underlying, f) with SeqView[B] {
     def apply(idx: Int): B = f(underlying(idx))
     def length: Int = underlying.length
   }
 
+  @SerialVersionUID(3L)
+  class Appended[+A](underlying: SomeSeqOps[A], elem: A) extends View.Appended(underlying, elem) with SeqView[A] {
+    def apply(idx: Int): A = if (idx == underlying.length) elem else underlying(idx)
+    def length: Int = underlying.length + 1
+  }
+
+  @SerialVersionUID(3L)
   class Prepended[+A](elem: A, underlying: SomeSeqOps[A]) extends View.Prepended(elem, underlying) with SeqView[A] {
     def apply(idx: Int): A = if (idx == 0) elem else underlying(idx - 1)
     def length: Int = underlying.length + 1
   }
 
+  @SerialVersionUID(3L)
+  class Concat[A](prefix: SomeSeqOps[A], suffix: SomeSeqOps[A]) extends View.Concat[A](prefix, suffix) with SeqView[A] {
+    def apply(idx: Int): A = {
+      val l = prefix.length
+      if (idx < l) prefix(idx) else suffix(idx - l)
+    }
+    def length: Int = prefix.length + suffix.length
+  }
+
+  @SerialVersionUID(3L)
+  class Reverse[A](underlying: SomeSeqOps[A]) extends AbstractIndexedSeqView[A] {
```

Comment:
I’m not sure this is a good idea to have `SeqView.Reverse`, because it relies on `reverseIterator`, which may consume the whole collection (in the case of an underlying `LinearSeq`, for instance).

Also, your definition extends `IndexedSeqView` but is located in `SeqView.scala`, I’m not sure what was your intent.

That being said, users can already call `.reverse` on a view, and this will consume the underlying collection (unless it’s an `IndexedSeq`). So, this `SeqView.Reverse` does not make things worse (it will actually benefit from optimizations applied to the underlying `reverseIterator` implementation). It’s just that I find a bit confusing to have a `View` that might deliberately consume more elements than those actually traversed by the user.

Reply:
It should extends `SeqView`. 

`reverse` already exists and it has to force a lazy collection and fully traverse a linear sequence when reading the first element. The only new part is that it now returns a `SeqView` as it should. If the source is sequential, so is the reversed view.

While `SeqView` provides methods for direct sequential access it does not imply that they are efficient, only that the collection is sequential. For efficient access you need `IndexedSeqView`.